### PR TITLE
Addition of an ExtensionEstimator

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request proposes an `ExtensionEstimator` based on the parameter estimator.

A possible improvement compared to the typical functionalities of the `ParameterEstimator` would be to handle a case where the extension parameter is kept fixed between all energies and then released to test the energy dependent morphology.

**Dear reviewer**

### CODE REVIEW

File: gammapy/estimators/extension.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            reoptimize=reoptimize,
+        )
+
+    def run(self, datasets):
+        """Run."""
+        datasets = Datasets(datasets)
+        # find extension parameter
+        model = datasets.models[self.source].spatial_model
+
+        if hasattr(model, "sigma"):
```
Comment (adonath) [inline gammapy/estimators/extension.py:91]: I think this code block should go into the `estimate_size()` function instead and use the size parameter just as local variable.

**Comment by adonath** [comment]
Thanks @registerrier, this is very interesting and covers an important use case. As this is still a draft I just have a few preliminary comments:

- Depending on the parametrization of the model there is typically a strong correlation between the amplitude and the size parameter. So I think the default behavior here should be to always re-fit the amplitude, when varying the size parameter.
- In general we could think about  having an (optional) `Map`, that stores the size parameter and size scan in `FluxMaps` as well.

**Comment by bkhelifi** [comment]
@registerrier, should you update your branch such that the tests pass?

File: gammapy/estimators/extension.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    The size can be estimated in various energy intervals.
+
+    Parameters
+    ----------
+    energy_edges : `~astropy.units.Quantity`
+        Edges of the energy intervals where the extension is estimated.
+    source : str or int
+        For which source in the model to compute the extension.
+    size_min : float
```
Comment (bkhelifi) [inline gammapy/estimators/extension.py:23]: Maybe the default values should be added in the docstring...

File: gammapy/estimators/extension.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        Fit instance specifying the backend and fit options.
+    reoptimize : bool
+        Re-optimize other free model parameters. Default is True.
+    """
+
+    tag = "ExtensionEstimator"
+
+    def __init__(
+        self,
+        energy_edges=[1, 10] * u.TeV,
```
Comment (bkhelifi) [inline gammapy/estimators/extension.py:54]: Maybe these default values should be removed... to be generic and to force the user to add edges

**Review by bkhelifi — COMMENTED** [review]
Beyond the few comments on the docstring and initial values, it seems OK to be merged...

**Comment by bkhelifi** [comment]
@registerrier can you update your code from the main?

**Comment by registerrier** [comment]
Closing for now. We need to think more generally how to work with morphology estimation.

**Pull request closed without merging**